tools/cmake/README.md

```diff
@@ -82,27 +82,56 @@ If you're running CMake directly, you may need to pass it in yourself as follows
 cmake -S <source-dir> -B <build-dir> -DCMAKE_BUILD_TYPE=debug
 ```
 
-## How to build a greentea test
-
-Install prerequisites suggested in the previous section and follow the below steps to build:
-* Set your current directory to the test suite directory
-
-* Run the following command for the configuration CMake module to be generated
-  ```
-  mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET> --mbed-os-path /path/to/mbed-os
-  ```
-* Build the test binary with the full profile
+## How to build and run greentea tests
+
+### Set up
+Install the [mbed-hosts-tests](https://pypi.org/project/mbed-host-tests/) python package.
+
+Also install the prerequisites suggested in the previous section.
+
+### Configuring the build system
+1. Set your current directory to the mbed-os root.
+1. Connect your target and detect it using `mbed-tools`
+   ```
+   mbedtools detect
+   ```
+   _Make a note of the mount point and serial port. We will need to pass them to CMake in a later step._
+
+1. Generate an mbed_config.cmake configuration file for your target using `mbed-tools`
+   ```
+   mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET> --mbed-os-path . --output-dir __build
+   ```
+1. Generate the build system with CMake, configuring it to build greentea tests.
+   Here you will pass the serial port and mount point you noted in an earlier
+   step
+   ```
+   cmake -S . -B __build -G Ninja -DBUILD_GREENTEA_TESTS=ON -DCMAKE_CTEST_ARGUMENTS="--output-on-failure;-V" -DMBED_HTRUN_ARGUMENTS="-p;<serial-port-from-detect>;-d;<mount-point-from-detect>"
+   ```
+   The `MBED_HTRUN_ARGUMENTS` argument is a semicolon separated list of arguments
+   to forward to `mbedhtrun`. You can pass any of the optional flags you would
+   usually pass to `mbedhtrun` by adding them to this list.
+
+   The `CMAKE_CTEST_ARGUMENTS` argument is a semicolon separated list of
+   arguments to forward to CTest, for more details see [the CMake documentation.](https://cmake.org/cmake/help/latest/variable/CMAKE_CTEST_ARGUMENTS.html)
+
+  If you want to build with the "bare-metal" profile, append
+  `-DMBED_GREENTEA_TEST_BAREMETAL=ON` to the above command.
+
+### Build and run tests
+  To build and run the tests run the following commands
   ```
-  cd cmake_build/<MBED_TARGET>/<PROFILE>/<TOOLCHAIN>/ && cmake ../../../.. -G Ninja && cmake --build .
+  cmake --build __build && cmake --build __build --target test
   ```
-  Or build the test binary with the baremetal profile
+  Alternatively you can run ctest directly
   ```
-  cd cmake_build/<MBED_TARGET>/<PROFILE>/<TOOLCHAIN>/ && cmake ../../../.. -G Ninja -DMBED_GREENTEA_TEST_BAREMETAL=ON && cmake --build .
+  cmake --build __build
+  cd __build
+  ctest -V
```

Comment:
Is it worth adding a note on how to run an individual test?

Reply:
I'll add a link to the ctest documentation to avoid repeating it here.

Reply:
Even though how to run an individual test is in the CTest documentation, it's still good to let users know how to _build_ one test. Presumably using `--target <test-name>` instead of `--target test`? Since building & testing of one test is a very commonly done task, giving a special mention of both in one place could be useful in my opinion. 

Especially that the build time for all tests has increased quite a lot by now (e.g. over 10K object files). Even though it's partially because `INTERFACE` libraries does lots of duplicated compilation than `STATIC` (a bit off-topic now...)